vagrant-spk

```diff
@@ -620,6 +673,125 @@ def dev(args):
         "spk dev --pkg-def=/opt/app/.sandstorm/sandstorm-pkgdef.capnp:pkgdef"
     ]))
 
+def inject_enter_grain_into_grain(sandstorm_dir, pid, enter_grain_checksum, enter_grain_binary,
+                                  tmp_vagrant_spk_dir, enter_grain_binary_path):
+    ### The purpose of this function is to take a PID, which is presumably the root process of a
+    ### grain within the vagrant-spk VM, and make sure that /tmp/vagrant-spk/enter_grain is the
+    ### binary that we want. If is isn't, then inject it.
+    sys.stderr.write("Adding enter_grain program to the grain in /tmp/vagrant-spk/enter-grain...\n")
+    tmp_vagrant_spk_dir = '/proc/{}/root/tmp/vagrant-spk'.format(pid)
+    enter_grain_binary_path = tmp_vagrant_spk_dir + '/enter-grain'
+    commands = ' && '.join([
+        # Make sure /tmp/vagrant-spk exists in the VM.
+        'sudo mkdir -p {}'.format(tmp_vagrant_spk_dir),
+        # If the sha1sum tool exists, and the binary in this directory has the right sha1sum,
+        # then skip sending it over.
+        'if which sha1sum >/dev/null && sudo sha1sum {} 2>/dev/null | grep -q ^{} ; then sudo chmod 755 {} && exit 0 ; fi'.format(
+            enter_grain_binary_path, enter_grain_checksum, enter_grain_binary_path),
+        # If not, then use cat to inject it in.
+        'sudo dd of={} 2>/dev/null'.format(enter_grain_binary_path),
```

Comment:
I'll fix the comment.

I use `dd` not `cat` because `dd` can open a file itself, and I need the file opened as root, so `sudo cat > filename` will give EPERM.